app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/MosaicImplicits.scala

```diff
@@ -113,127 +230,77 @@ class MosaicImplicits[HistStore: HistogramStore](histStore: HistStore)
               IO.raiseError(NoDataInRegionException)
             } else IO.unit
           }
-          filtered = BacksplashMosaic.filterRelevant(self)
-          extent = BacksplashImage.tmsLevels(z).mapTransform.keyToExtent(x, y)
+          overviewStream = self zip {
+            self flatMap { (backsplashIm: BacksplashImage[IO]) =>
+              fs2.Stream.eval(
+                projStore.getOverviewLocation(backsplashIm.projectLayerId)
+              )
+            }
+          } take (1) evalMap {
+            case (im, loc) =>
+              loc map { uri =>
+                IO.pure(
+                  BacksplashGeotiff(im.imageId,
+                                    im.projectId,
+                                    im.projectLayerId,
+                                    uri,
+                                    im.subsetBands,
+                                    im.corrections,
+                                    im.singleBandOptions,
+                                    im.mask,
+                                    im.footprint))
+              } getOrElse {
+                IO.raiseError(
+                  new MetadataException("No overview location found"))
+              }
+          }
           // for single band imagery, after color correction we have RGBA, so
           // the empty tile needs to be four band as well
           mosaic <- if (bandCount == 3) {
-            val ioMBT = filtered
-              .parEvalMap(streamConcurrency)({ relevant =>
-                for {
-                  imFiber <- {
-                    logger.debug(
-                      s"Reading Tile ${relevant.imageId} ${z}-${x}-${y}")
-                    relevant.read(z, x, y)
-                  }.start
-                  histsFiber <- {
-                    logger.debug(
-                      s"Reading Histogram ${relevant.imageId} ${z}-${x}-${y}")
-                    getHistogramWithCache(relevant)
-                  }.start
-                  hists <- histsFiber.join
-                  im <- imFiber.join
-                  renderedTile <- IO.pure {
-                    im map { mbTile =>
-                      val noDataValue = getNoDataValue(mbTile.cellType)
-                      logger.debug(
-                        s"NODATA Value: ${noDataValue} with CellType: ${mbTile.cellType}"
-                      )
-                      relevant.corrections.colorCorrect(mbTile,
-                                                        hists,
-                                                        noDataValue)
-                    }
-                  }
-                } yield {
-                  renderedTile
-                }
+            val filtered = BacksplashMosaic.filterRelevant(self)
+            if (z <= Config.tuning.overviewThreshold) {
```

Comment:
I don't think this should be a configuration, there is a value `minZoomLevel` on `ProjectLayer` that should be used instead.

Reply:
I did not realize such a variable existed.

Reply:
I know I'm the author of the issue, but a lot of this would have been really helpful to include in the issue :man_facepalming: